profiles/tests.py

```diff
@@ -230,3 +239,195 @@ def test_save_user_profile_when_profile_exists(self):
         # Assert that the existing UserProfile is not affected
         self.assertIsNotNone(user_profile)
         self.assertEqual(user_profile.user, user)
+
+
+class SendFriendRequestTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            username="testuser", password="testpassword"
+        )
+        self.friend = User.objects.create_user(
+            username="testcreator", password="testpassword"
+        )
+        if not hasattr(self.user, "userprofile"):
+            # Create a new user profile
+            self.user_profile = UserProfile.objects.create(
+                user=self.user,
+                bio="Test Bio",
+                # Add other required fields as needed
+            )
+        else:
+            # Use the existing user profile
+            self.user_profile = self.user.userprofile
+        if not hasattr(self.friend, "userprofile"):
+            # Create a new user profile
+            self.friend_profile = UserProfile.objects.create(
+                user=self.friend,
+                bio="Test Bio",
+                # Add other required fields as needed
+            )
+        else:
+            # Use the existing user profile
+            self.friend_profile = self.friend.userprofile
+        self.client = Client()
+
+    def test_toggle_friend_request_create(self):
+        self.client.login(username="testuser", password="testpassword")
+        url = reverse("profiles:toggle-friend-request", args=[self.friend_profile.id])
+        # Initially, the user has not joined the event
+        response = self.client.post(url)
+        # Check that the EventJoin was created with the status 'pending'
+        join = UserFriends.objects.get(user=self.user, friends=self.friend_profile)
+        self.assertEqual(join.status, PENDING)
+        # Make the POST request again to toggle the status to 'withdrawn'
+        response = self.client.post(url)
+        # Fetch the updated join object and check its status
+        join.refresh_from_db()
+        self.assertEqual(join.status, WITHDRAWN)
+        self.client.post(url)
+        join.refresh_from_db()
+        self.assertEqual(join.status, PENDING)
+        # Check the response to ensure the user is redirected to the event detail page
+        self.assertRedirects(
+            response, reverse("profiles:view_profile", args=[self.friend_profile.id])
+        )
+
+    def test_user_cannot_join_own_event(self):
+        self.client.logout()
```

Comment:
Same here, if we can replace all the "join" related naming that will be great. Thank you!

Reply:
Made the change

Reply:
Sorry for being a bit strict here, just trying to be a bit cleaner. Could you also change the naming in the def test_toggle_friend_request_create? Thank you! Otherwise I think it's good!

Reply:
Do you mean the comments?
I've changed the comments for that